### DISCUSSION TITLE
How to change collation of database from utf8mb4_0900_ai_ci to utf8mb4_general_ci?

### DISCUSSION THREAD

**Discussion by jeffminsungkim** [planetscale/discussion#337] [Q&A] [answered]
I need to change the collation to get an accurate search result which is in Korean. I've tried the following command lines below, but it didn't work.

```sql
ALTER DATABASE <database_name> CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;

ALTER TABLE <table_name> CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;
```

Does PlanetScale only support `utf8mb4_0900_ai_ci`?

**Comment by fooforge** [comment, marked as answer]
Hi @jeffminsungkim,

PlanetScale [supports all](https://planetscale.com/docs/reference/mysql-compatibility#overview) `utf8`, `utf8mb4` and `utf8mb3` character sets, and we create new databases with the `utf8mb4_0900_ai_ci` collation. As we do not support the `ALTER DATABASE` statement you will not be able to change your database's default collation, though. You can change it on a per-table basis, however.

I have checked your database's schema and can confirm one of your tables now has the `utf8mb4_general_ci` collation so your `ALTER TABLE` statement seems to have worked as intended. See for yourself by executing `SHOW TABLE STATUS\G;`.

I suspect your `ALTER TABLE` query did not throw an error after all and that you've got this mixed up with the error thrown by the `ALTER DATABASE` query. Let me know, though, if you did in fact see an error message and we'll look into it further.

**Reply by jeffminsungkim** [reply]
Thanks for the response @fooforge.
Yeah I've read the documentation and tried alter a single table to make the collation to `utf8mb4_general_ci` and it worked. Thought there was no other options for me.
I feel sorry that PlatnetScale does not allow to change the db collation itself. I should stick with on a per-table basis instead.

Thank you again. :)
